bundles/go/go_fmt.moon

```diff
@@ -0,0 +1,54 @@
+import app, config, mode, signal from howl
+import Process from howl.io
+
+append = table.insert
+
+run_command = (contents) ->
+  args = {}
+  for arg in config.go_fmt_command\gmatch '%S+'
+    append args, arg
+  args.stdin = contents
+  success, out, err, p = pcall Process.execute, args, stdin: contents
+  unless success
+    log.error out
+    return nil
+  unless p.successful
+    log.error err
+    return nil
+  out
+
+calculate_new_pos = (pos, before, after) ->
+  new_pos = 1
+  i = 1
+  while i <= pos and new_pos <= #after
+    if before[i] == after[new_pos]
+      i += 1
+      new_pos += 1
+    elseif before[i]\match '%s'
+      i += 1
+    else
+      new_pos += 1
+  new_pos-1
+
+fmt = (buffer) ->
+  log.info "Running #{config.go_fmt_command}..."
+  before = buffer.text
+  after = run_command before
+  log.info "#{config.go_fmt_command} completed"
+  return if not after or after == before
+
+  editor = app\editor_for_buffer buffer
+  if editor
+    -- reload the contents, adjusting position
```

Comment:
It seems like `Editor:with_position_restored` could be used instead of doing our own custom thing here for the positioning?